### PR TITLE
TICK-355 Separate ProxSlope from ProxSortedL1 

### PR DESCRIPTION
**Pull request by stephanegaiffas**
Separate ProxSlope from ProxSortedL1 (C++ and python). 
This will be useful to : 
* provide easy features selection in inference tools, by enabling 'slope' as penalization 
* incorporate Alain's PhD work about outliers detection

### CODE REVIEW

**Comment by stephanegaiffas** [comment]
@Mbompr : ready for review

File: tick/optim/prox/src/prox_sorted_l1.cpp [hunk 1]
```diff
@@ -8,173 +8,149 @@
...
-        sub_out[idx[i]] = subsub_out[i];
+      sub_coeffs_sign[i] = -1;
+      sub_coeffs_abs[i] = -sub_coeffs_i;
     }
-    for (ulong i = 0; i < size; i++) {
-        sub_out[i] *= sub_coeffs_sign[i];
+  }
+  // Where do the crossing occurs?
+  ulong crossing = 0;
+  for (ulong i = 0; i < size; i++) {
```
Comment (vinther) [inline tick/optim/prox/src/prox_sorted_l1.cpp]: Could we iterate over this in reverse order to get the crossing faster?

File: tick/optim/prox/src/prox_sorted_l1.cpp [hunk 2]
```diff
@@ -8,173 +8,149 @@
...
     }
+  }
+  ulong n_sub_coeffs;
+  if (crossing > 0) {
+    n_sub_coeffs = crossing + 1;
+  } else {
+    n_sub_coeffs = size;
+  }
+
+  ArrayDouble subsub_coeffs = view(sub_coeffs_abs_sort, 0, n_sub_coeffs);
```
Comment (vinther) [inline tick/optim/prox/src/prox_sorted_l1.cpp]: What happens if `crossing == size`? Then `n_sub_coeffs == size + 1 `and it looks like we could go out of bounds.

**Comment by stephanegaiffas** [comment]
@vinther : can you finish the review ?

**Comment by vinther** [comment]
Done. I can rebase and merge if you want.

**Pull request merged**